src/vivarium_census_prl_synth_pop/components/household.py

```diff
@@ -138,65 +139,50 @@ def on_time_step(self, event: Event):
         """
         households = self.population_view.get(event.index)
         household_heads = households.loc[
-            households["relation_to_household_head"] == "Reference person"
-        ]
-        households_that_move = self.addresses.determine_if_moving(
-            household_heads["household_id"], self.household_move_rate
+            households["relation_to_household_head"] == "Reference person"]['household_id']
+
+        households_that_move_idx = self.randomness.filter_for_rate(
+            household_heads.index, self.household_move_rate(household_heads.index)
         )
 
-        # Find households that move abroad and separate subsets of state table
-        households_that_move_abroad_idx = self.determine_households_moving_abroad(
-            households_that_move, self.proportion_households_leaving_country, event
+        # Determine which households move abroad
+        households_heads_that_move_abroad_idx = filter_by_rate(
+            households_that_move_idx, self.randomness, self.proportion_households_leaving_country, "abroad_households"
         )
-        moving_abroad_households = households_that_move.loc[households_that_move_abroad_idx]
-        moving_domestic_households = households_that_move.loc[
-            ~households_that_move.index.isin(moving_abroad_households.index)
-        ]
-        abroad_moving_households = households.loc[
-            households["household_id"].isin(moving_abroad_households)
-        ]
-        domestic_moving_households = households.loc[
-            households["household_id"].isin(moving_domestic_households)
+        # Find households_ids that move abroad and domestic
+        abroad_households_ids = household_heads.loc[households_heads_that_move_abroad_idx]
+        domestic_household_ids = household_heads.loc[
```

Comment:
You can use set difference functions on indices to get the index you want here which I think is a bit more readable when you're manipulating Indexes.
```
 domestic_household_ids = household_heads.loc[
    household_heads.index.isin(
        households_that_move_idx.difference(households_heads_that_move_abroad_idx)
    )
]
```